### PR TITLE
PostAgent can emit Events

### PR DESCRIPTION
**Pull request by cantino**
This PR allows PostAgents to emit Events (initial work on #1206).

@flechaig, would you be interested in testing this out?

### CODE REVIEW

**Comment by cantino** [comment]
@dsander, mind a quick review?

File: app/models/agents/post_agent.rb [hunk 1]
```diff
@@ -23,7 +25,17 @@ class PostAgent < Agent
         * `user_agent` - A custom User-Agent name (default: "Faraday v#{Faraday::VERSION}").
     MD
 
-    event_description "Does not produce events."
+    event_description <<-MD
+      Events look like this:
+        {
+          "status": 200,
+          "response_headers": {
```
Comment (dsander) [inline app/models/agents/post_agent.rb]: Maybe call it `headers` to make it similar to the `WebsiteAgent`?

File: app/models/agents/post_agent.rb [hunk 2]
```diff
@@ -23,7 +25,17 @@ class PostAgent < Agent
...
+            "Content-Type": "text/html",
+            ...
+          },
+          "body": "<html>Some data...</html>"
+        }
+    MD
 
     def default_options
```
Comment (dsander) [inline app/models/agents/post_agent.rb]: Would it make sense to add `emit_events: false` to the default options?

File: app/models/agents/post_agent.rb [hunk 3]
```diff
@@ -56,6 +68,10 @@ def validate_options
         errors.add(:base, "if provided, payload must be a hash")
       end
 
+      if options['emit_events'].present? && ![true, false, 'true', 'false'].include?(options['emit_events'])
```
Comment (dsander) [inline app/models/agents/post_agent.rb]: How about ` if options['emit_events'].present? && boolify(options['emit_events']).nil?`?

**Comment by dsander** [comment]
Neat, looks good to me otherwise.

File: app/models/agents/post_agent.rb (hunk 1 above)
Comment (cantino) [inline app/models/agents/post_agent.rb]: That's reasonable.

File: app/models/agents/post_agent.rb (hunk 3 above)
Comment (cantino) [inline app/models/agents/post_agent.rb]: Sure, that's cleaner.

**Comment by cantino** [comment]
Thanks!

File: app/models/agents/post_agent.rb (hunk 2 above)
Comment (cantino) [inline app/models/agents/post_agent.rb]: Done.

**Pull request merged**